soh/soh/Enhancements/item-tables/ItemTableTypes.h

```diff
@@ -6,14 +6,22 @@
 #define CHEST_ANIM_SHORT 0
 #define CHEST_ANIM_LONG 1
 
+/**
+ * Flag to indicate which type of Actor has given the player an item. ITEM_FROM_NPC by default,
+ * should be set by the actor giving the item before they send the GetItemEntry signal to the player.
+ */
+typedef enum GetItemFrom {
+    ITEM_FROM_NPC,
+    ITEM_FROM_SKULLTULA,
+    ITEM_FROM_FREESTANDING,
+    ITEM_FROM_CHEST,
+} GetItemFrom;
+
 #define GET_ITEM(itemId, objectId, drawId, textId, field, chestAnim, modIndex, getItemId) \
-    { itemId, field, (chestAnim != CHEST_ANIM_SHORT ? 1 : -1) * (drawId + 1), textId, objectId, modIndex, getItemId, drawId, true, NULL }
+    { itemId, field, (chestAnim != CHEST_ANIM_SHORT ? 1 : -1) * (drawId + 1), textId, objectId, modIndex, getItemId, drawId, true, ITEM_FROM_NPC, NULL }
```

Comment:
just curious here, why default to `ITEM_FROM_NPC`?

Reply:
Because that required the least amount of changes to existing actor files to add the ITEM_FROM value.

Reply:
sounds reasonable, might be nice to add a little comment explaining that decision

Reply:
There is a bit of an explanation in a comment above the `GetItemFrom` enum, but I'll expand that a bit.

Reply:
the updated comment is perfect!